src/ncurses_ui.cc

```diff
@@ -1361,6 +1368,22 @@ void NCursesUI::set_ui_options(const Options& options)
         m_wheel_scroll_amount = wheel_scroll_amount_it != options.end() ?
             str_to_int_ifp(wheel_scroll_amount_it->value).value_or(3) : 3;
     }
+
+    {
+        auto it = options.find("ncurses_padding_char"_sv);
+        if (it == options.end())
+            m_padding_char = DisplayAtom("~");
+        else if (it->value.length() < 1)
```

Comment:
The point of this specific condition is just to avoid empty strings.
However I tested with strings and double-width characters and it works well (thanks to ncurses, I guess? Screenshots below).

Using `set-option -add global ui_options ncurses_padding_char=╱╳`
![screen01](https://user-images.githubusercontent.com/285009/115144339-753fb480-a04c-11eb-9771-defeda0d2b1e.png)

Using `set-option -add global ui_options ncurses_padding_char=奯`
![screen02](https://user-images.githubusercontent.com/285009/115144340-75d84b00-a04c-11eb-9005-934b4da8ade9.png)

Reply:
Ok, then we should at least take the `column_length()` of that string into account instead of the plain length and apply it line 480 to avoid pushing to much unused data past the end of the line.

Reply:
I don't see how to implement a past-the-column guard without a additional conditional branching. Would it be worth it? I guess the `draw` function is somehow speed bounded (?). The added value would be what? Smaller memory footprint for rare use-cases where the user configure long strings for the padding. In that case, wouldn't it be a better compromise to limit the column length of the option?

Reply:
`it->value.length()` here returns the byte length of the string, I think that for what you are checking it makes more sense to look at `it->value.column_length()` which will catch zero-width codepoints and avoid corner cases with the suggested change on line 480.

Reply:
I understand now, thanks. Additional commit done.